src/nspawn/nspawn.c

```diff
@@ -2990,6 +2993,13 @@ static int outer_child(
         if (mount(directory, directory, NULL, MS_BIND|MS_REC, NULL) < 0)
                 return log_error_errno(errno, "Failed to make bind mount: %m");
 
+        /* Mark everything as shared so our mounts get propagated down. This is
+         * required to make new bind mounts available in systemd services
+         * inside the containter that create a new mount namespace.
+         * See https://github.com/systemd/systemd/issues/3860 */
```

Comment:
Add a further comment:

Further submounts (such as /dev) done after this will inherit the shared propagation mode.